### PR TITLE
feat: install linter formatter

### LINKED ISSUE #4: Install linter & formatter
## Description

- This task is to install and configure the linter and formatter to achieve consistent code quality by setting up Husky.
- This task depends on #3 .

## Criteria

- [x] Install [Prettier](https://prettier.io/) with exact version
- [x] Install [CommitLint](https://commitlint.js.org/)
- [x] Install [Lint-staged](https://github.com/lint-staged/lint-staged#readme)
- [x] Install [Husky](https://typicode.github.io/husky/)
- [x] Configure Prettier 
- [x] Configure ESLint
- [x] Configure CommitLint
- [x] Configure pre-commit hooks to make it run these linter & formatter checks after a commit is made, with Husky and Lint-staged
- [x] Create scripts in `package.json` to run each formatter/linter
- [x] Format all the files at the end

## Notes

- It might be helpful to take a look at [.prettierrc](https://github.com/Tascurator/tascurator-documents/blob/main/.prettierrc) in `Tascurator/tascurator-documents` repository as for the prettier configuration file.
- Also, it is a good practice to prevent folders like `node_modules` from being formatted by Prettier.

### PR DESCRIPTION
**Pull request by KaiKoide**
## Overview

To facilitate smooth development, set up the linter and formatter to main consistent code quality in the Next.js + TypeScript development environment.

## Changes

- [x] Install [Prettier](https://prettier.io/)
- [x] Install [CommitLint](https://commitlint.js.org/)
- [x] Install [Lint-staged](https://github.com/lint-staged/lint-staged#readme)
- [x] Install [Husky](https://typicode.github.io/husky/)
- [x] Configure Prettier
- [x] Configure ESLint
- [x] Configure CommitLint
- [x] Configure pre-commit hooks to make it run these linter & formatter checks after a commit is made, with Husky and Lint-staged
 
## Assignee Checklist:

- [x] The base branch is correct (no accidental merges)
- [x] The branch name follows our branch naming rules
- [x] The PR title follows our PR title rules
- [x] My code follows our coding style
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings or errors in the console

## Reviewer Checklist

- [x] The code will be formatted before committing
- [x] Invalid commit messages cannot be committed

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,22 +6,37 @@
...
-    "lint": "next lint"
+    "lint": "next lint",
+    "format": "prettier . --write",
+    "format-file-patterns": "prettier \"./src/**/*.{js,jsx,ts,tsx,json,css}\" --write",
+    "format-ignore-path": "prettier . --write --ignore-path {any file}",
+    "prepare": "husky"
+  },
+  "lint-staged": {
+    "**/*.{js,jsx,ts,tsx,json,css,scss,html,md}": "prettier --write"
   },
```
Comment (akiomatic) [inline package.json]: I'd like you to revise the pre-commit script to run ESLint checks as well as Prettier checks.

File: package.json [hunk 2]
```diff
@@ -6,22 +6,37 @@
     "dev": "next dev",
     "build": "next build",
     "start": "next start",
-    "lint": "next lint"
+    "lint": "next lint",
+    "format": "prettier . --write",
+    "format-file-patterns": "prettier \"./src/**/*.{js,jsx,ts,tsx,json,css}\" --write",
+    "format-ignore-path": "prettier . --write --ignore-path {any file}",
```
Comment (akiomatic) [inline package.json]: I'd like you to add scripts to run ESLint checks like `lint:fix` that fixes ESLint errors automatically.

**Comment by akiomatic** [comment]
Hi! Could you change the PR title to the appropriate one?

File: .prettierignore
Comment (akiomatic) [inline .prettierignore:1]: Could you add the following to avoid unnecessary formatting?

```
node_modules/
.next/
```

**Review by akiomatic — CHANGES_REQUESTED** [review]
Thank you for setting up the linter & formatter, but I kindly ask you to take a look at my comments.

File: package.json (hunk 2 above)
Comment (KaiKoide) [reply package.json]: I add `"lint:fix": "eslint --fix ."` to `package.json`.

File: package.json (hunk 1 above)
Comment (KaiKoide) [reply package.json]: I add `lint:fix` in `package.json`.

File: .prettierignore
Comment (KaiKoide) [reply .prettierignore:1]: I updated `.prettierignore`.

**Review by akiomatic — APPROVED** [review]
LGTM!
Thank you for setting up!

**Review by Akikaze1119 — APPROVED** [review]
Thank you!
I've done some testing and it looks fine.

**Review by HrRn — APPROVED** [review]
Thank you!

**Pull request merged**